Merge 'Development' branch into 'Main' branch - Datalink 2.0.0

4x8Matrix opened the pull request:
# Context
Datalink has, and had needed a large update refactoring the technology being used; the source code behind Datalink being all too a bit too obscure and had no designated style-guide. 

The goal of this PR is to introduce a layer of understandability into the source, enabling those who want to contribute or add in features, to do so (or suggest them — We will listen!)

---

I'd go as far as saying this is the official 2.0.0 for Datalink; Datalink 1.0.0 is classed as the Prototype version, used for testing, regulating and essentially helping us developers find our feet.

# This PR
**Please refer to the 'Commit-History' feature as this PR is very large**

Commit:
🧨 Remove deprecated source code

Commit:
🔧 Update development tools & workflows

Commit:
⬆️ Added git SSH submodules

Commit:
✅ Add /test directory & runner

Commit:
♻️ Implementation of refactored SDK

Commit:
✨ TypeScript Datalink Support

Commit:
✨ Implementation of Toolchain TOML files

Commit:
♻️ Refactor GIT readme

Commit:
🧨 Remove outdated/old programming

Commit:
♻️ Updated Datalink project structure

Commit:
✨ Implementation of the 'Signal' submodule

Commit:
✨ Imlementation of 'Enum' modules

Commit:
✨ Imlementation of 'Data' modules

Commit:
✅ Addition of TestEZ tests

Commit:
♻️ Refactored module interface - /source

Commit:
♻️ Refactored HTTP module - /source/Components/HttpComponent

Commit:
✨ Implementation of 'Daemon' Component

Commit:
✨ Implementation of 'Scheduler' Component

Commit:
✨ Implementation of 'Session' Component

Commit:
✨ Implementation of 'Throttle' Component

Commit:
Merge branch 'main' of github.com:datalinkhq/rbx-datalink into development

CompeyDev commented:
Please follow [this](https://gist.github.com/5fc875a6e45844040ad9d9a3215f94e5) style guide, as the project will need to be bootstrapped with roblox-ts once again. The current style guide applies to lua only. 

Also, you might want to sign your commits next time onwards using GPG before pushing them to git.

CompeyDev commented:
**Do not** merge this PR, as I will be setting my TypeScript compatibility soon.

CompeyDev committed:
:sparkles: Initial roblox-ts managed project

Commit:
fix: actions workflow

TODO: Fix unexpected EOT syntax error

CompeyDev committed:
:twisted_rightwards_arrows: rebase & merge

CompeyDev committed:
:green_heart: fix CI (2)

I guess you cannot use the same value for two keys in yaml like a case
switch. I learn something every day!

CompeyDev committed:
:green_heart: fix CI (4)

Maybe you can have duplicate key values using a shorthand? Let's try
this.

CompeyDev committed:
:green_heart: fix CI (5)

CompeyDev commented on .github/workflows/Selene.yml:
```diff
@@ -1,23 +1,27 @@
 name: Selene Linter
```
Will be redoing the linter, because the End of Lines seem corrupted.

CompeyDev reviewed (approved):
Looks good, I'll make some changes and we'll be on our way to merge.

4x8Matrix commented:
> Please follow [this](https://gist.github.com/5fc875a6e45844040ad9d9a3215f94e5) style guide, as the project will need to be bootstrapped with roblox-ts once again. The current style guide applies to lua only.

I disagree, you should try to follow the [Roblox LuaU styleguide](https://roblox.github.io/lua-style-guide/) as that is the core languages of this SDK, me using typescript favors' the majority which shouldn't be the case. 

That said ~ the StyleGuide linked is also not documented well as it covers two subjects..? 

> Also, you might want to sign your commits next time onwards using GPG before pushing them to git.

My bad, moved to arch so my GPG keys got invalidated, will update them soon! 

> Do not merge this PR, as I will be setting my TypeScript compatibility soon.

Thats fine, please don't approve until you're happy the PR can be merged.

Commit:
✨ Update 'Data' modules

Commit:
✨ Update 'Enum' modules

Commit:
✅ Addition of 'PublishEvent' test

Commit:
✨ Implementation of UTC Date Component

Commit:
🧨 Deprecate old code

Commit:
✨ Implementation of 'fireCustomEvent' function

Commit:
⬆️ Update SIFT Commit Hash

Commit:
Merge branch 'development' of github.com:datalinkhq/rbx-datalink into development

Commit:
♻️ Refactor proj structure - minor merge conflicts

Commit:
♻️ Minor adjustments to match DOCS

Commit:
chore(licensing): move to affero GNU license

Commit:
merge: PR #2 from datalinkhq/licensing-agpl

Licensing: Relicense to AGPL

Commit:
chore(licensing): remove MIT

Commit:
chore(licensing): use markdown instead of plaintext

Commit:
✅ Updated tests

Commit:
✨ Addition of 'getPlayerHash' method

Commit:
✨ Addition of 'SessionParameters' Enum

Commit:
✨ Addition of 'PlayerComponent'

Commit:
git pushMerge branch 'development' of github.com:datalinkhq/rbx-datalink into development

Commit:
✨ Update 'DatalinkInstance' type

Commit:
✨ Updated 'PublishLog' and 'FetchLog'

Commit:
✨ Updated 'PublishLog' and 'FetchLog' Data

Commit:
♻️ Updated 'DateComponent' language from LuaU to Lua

Commit:
✨ Implementation of 'getGameLogAsync' and 'getAllGameLogsAsync' methods

Commit:
✅ Addition of 'Logging' test

Commit:
✨ Implementation of 'LoggerComponent'

Commit:
✅ Updated FOCUS for Authentication Test

Commit:
✨ Added usage of Datalink Janitor

Commit:
🚨 Selene Lint error fix

Commit:
✨ Added Typed support for new methods

Commit:
✨ Addition of the 'onHeartbeat' signal

Commit:
✨ Invoke 'onHeartbeat' signal once heartbeat is invoked

Commit:
✨ Invoke 'onThrottled' once Datalink is either throttled or unthrottled

Commit:
✨ Invoke 'onMessageRequestSent' & 'onMessageRequestFail' when a message sent has failed or succeeded

Commit:
🐛 Signal impl incorrect; Updated signal reference

Commit:
✨ Implementation & Invoke of 'onDaemonInitiated' for Daemons used under Datalink SDK

CompeyDev committed:
feat: initial roblox-ts compatible TypeScript types

howmanysmall commented:
🤓🤓🤓🤓🤓🤓🤓🤓🤓🤓🤓
![NerdEmojiNerdGIF](https://user-images.githubusercontent.com/26746527/209567044-16e97a8b-ac75-4dc4-8368-4c461ac4bb68.gif)
🤓🤓🤓🤓🤓🤓🤓🤓🤓🤓

Commit:
✨ Addition of 'SDK-Interfaces'

Commit:
✨ Updated 'Enumeration' project files

Commit:
✨ Updated 'Data' project files

Commit:
✨ Addition of 'TeleportComponent' to handle player teleports

Commit:
✨ Refactored structure for project components

Commit:
✨ Addition of 'Types' module

Commit:
fix(grammar): we use a `:` not a `;`

Semicolons are use to denote sentence closure without actually closure. Listing is done with a colon.

Commit:
🧨 Removal of 'Sha256' version

Commit:
✨ Implementation of refactored Sha256 Component

Commit:
👀 Update server with local changes

Commit:
✨ Deprecate and take advantage of new Sha 256 Component

Commit:
✨ Update implementation of 'Sha256' algorithm

Commit:
✨ Update 'src' -> 'source'

Commit:
🧨 Deprecate outdated 'Test' folder

Commit:
✨ Implementation of new datalink 'Build' project

Commit:
✨ Addition of 'Event.spec.lua' to test 'Event' interface

Commit:
✨ Implementation of new 'Tests' system

Commit:
Merge branch 'development' of github.com:datalinkhq/rbx-datalink into feature/sdk-interfaces

Commit:
✨ Updated 'LoggerComponent' to respect published logs

4x8Matrix committed:
Merge pull request #4 from datalinkhq/feature/sdk-interfaces

Feature/sdk-interfaces: Implementation for a scalable project

Commit:
🧨 Remove broken 'src' folder

CompeyDev commented:
Everything is okay, however we should really consider using the TypeScript Style Guide since this repo is linted and managed primarily by `roblox-ts`.

Commit:
chore(rojo): roblox-ts typescript build configs

Commit:
chore(rojo) rename `ts.project.json` to `build_ts.project.json`

Commit:
chore(rbxtsc): updated project root dir from `src` to `source`

Commit:
🧨 Remove account Id & Token from 'DatalinkTestEZ' file

Commit:
🚨 Update selene lint config for 'global_usage'

Commit:
💚 Fix 'Selene' lint workflow

Commit:
💚 Update 'Spaces' -> 'Tabs'

Commit:
💚 Updated selene workflow 'branches'

Commit:
💚 Update 'Tabs' -> 'Spaces' ??

Commit:
✨ Add 'build' tasks for VSCode

CompeyDev reviewed (approved):
Ooh, VSCode tasks... That must be helpful for new contributors!

Commit:
:green_heart: Use aftman instead of foreman

No aftman configs are provided, hence the CI fails.

Commit:
:green_heart: fix aftman version

Commit:
:green_heart: remove release workflow for the timebeing

Commit:
:green_heart: use fixed version, `setup-aftman` does not support globs

Commit:
:green_heart: aftman releases are prefixed with a "v"

Commit:
chore(README): fix docs image link

Commit:
chore(README): updated docs link

Commit:
chore(package): included meta fields

Commit:
Merge branch 'main' of github.com:datalinkhq/rbx-datalink into development
